File: lib/languages/zig/assignments.zig
// Variable declaration and assignment
const name = "Alice"; // Immutable
var age: u32 = 30;    // Mutable

// Mutability
age = 31;             // OK
// name = "Bob";      // Error: cannot assign to constant

// Explicit undefined initialization
var buffer: [100]u8 = undefined;

// Comptime variables (evaluated at compile time)
comptime var counter = 0;
counter += 1;

// Destructuring assignment
const point = .{ .x = 10, .y = 20 };
const x = point.x;
const y = point.y;

// Multiple returns and assignment
const values = getValues();
const a = values[0];
const b = values[1];

fn getValues() [2]i32 {
    return .{ 1, 2 };
}

// Pointer assignment
var value: i32 = 42;
var ptr: *i32 = &value;
ptr.* = 100;  // value is now 100

File: lib/languages/rust/assignments.rs
// Variable declaration and assignment
let name = String::from("Alice");
let age = 30;

// Mutability must be explicit
let mut counter = 0;
counter += 1;

// Immutable by default
// name = String::from("Bob");  // Error: cannot assign twice to immutable variable

// Shadowing (creating a new variable with the same name)
let value = 5;
let value = value * 2;  // value is now 10

// Destructuring assignment
let (x, y) = (1, 2);
let [head, tail @ ..] = [1, 2, 3]; // head = 1, tail = [2, 3]

// Destructuring structs
struct Point { x: i32, y: i32 }
let point = Point { x: 10, y: 20 };
let Point { x, y } = point;  // x = 10, y = 20

// Ignoring values
let (_, only_y) = (1, 2);  // only_y = 2

